### PR TITLE
Enabling fullscreen video in the internal browser

### LINKED ISSUE #504: Fullscreen video in "Internal Browser" does not work
Before I describe this some info on my setup:
Gif Viewer: Internal viewer
Video Viewer: Internal VideoView
Use Internal Browser: Enabled
ROM: CyanogenMod 6.0.1 based

Any video not supported by RedReader is passed to the internal browser. This is handled as far as I know by **Android System WebView** (I have the official Google one from the Play store installed).
The fullscreen option is available but when I press it nothing happens.

I found someone describing a similar problem for a different app:
https://teamtreehouse.com/community/html5-video-wont-go-fullscreen-in-webview

and he links to a possible solution:
https://stackoverflow.com/questions/15768837/playing-html5-video-on-fullscreen-in-android-webview/16179544#16179544

### PR DESCRIPTION
**Pull request by Clubfan22**
fixes #504
Thanks to @japanesephundroid for providing a link to the solution!

I ported @cprcrack's work on video enabled WebViews to the
current implementation, the code was taken from
https://github.com/cprcrack/VideoEnabledWebView .
I successfully tested viewing video both normally and in fullscreen
in OxygenOS Beta 25, which runs on Android 8.0.
Due to emulator problems, I wasn't able to test any other version of
Android, so I think this PR has to be tested thoroughly before merging,
especially on custom ROMs and different phone OEMs' implementations.

### CODE REVIEW

File: src/main/java/org/quantumbadger/redreader/views/webview/VideoEnabledWebChromeClient.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
+package org.quantumbadger.redreader.views.webview;
```
Comment (QuantumBadger) [inline src/main/java/org/quantumbadger/redreader/views/webview/VideoEnabledWebChromeClient.java]: This file needs a GPL notice :)

**Comment by QuantumBadger** [comment]
Thanks! I've left a couple of minor comments, but generally this looks fine to me.

File: src/main/java/org/quantumbadger/redreader/views/webview/WebViewFixed.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+// Also taken from cprcrack/VideoEnabledWebView
+// https://github.com/cprcrack/VideoEnabledWebView/blob/master/app/src/main/java/name/cpr/VideoEnabledWebView.java
+public class WebViewFixed extends WebView {
+
+	public class JavascriptInterface
+	{
+		@android.webkit.JavascriptInterface @SuppressWarnings("unused")
+		public void notifyVideoEnd() // Must match Javascript interface method of VideoEnabledWebChromeClient
+		{
+			Log.d("___", "GOT IT");
```
[the remark this answers is not in the document]
Comment (Clubfan22) [reply src/main/java/org/quantumbadger/redreader/views/webview/WebViewFixed.java]: That was in the original code, I'm going to remove it with my next commit though :)

**Comment by QuantumBadger** [comment]
Thanks, merging in!

**Comment by Clubfan22** [comment]
Should be fixed, and sorry for the blunders...

**Comment by QuantumBadger** [comment]
Not a problem!

Changelog updated here: https://github.com/QuantumBadger/RedReader/commit/095de5534a98e383724849c60fe3d323d51b30e9

Should be in the next Alpha in a few minutes :)

**Comment by japanesephundroid** [comment]
Can confirm that it works on stock LG Android 7.0 :)
Thanks guys.

**Pull request merged**